From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Tue, 27 Oct 2015 23:40:28 +0100
Subject: CMakeLists updated

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19ef8a5..e55b414 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,5 +63,9 @@ target_link_libraries(08_RMLVelocitySampleApplication Reflexxes)
 
 install(FILES ${REFLEXXES_INCLUDES} DESTINATION include/RML)
 install(TARGETS Reflexxes DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Reflexxes)
-
+install(TARGETS 01_RMLPositionSampleApplications 02_RMLPositionSampleApplications
+		03_RMLPositionSampleApplications 04_RMLVelocitySampleApplication 
+		05_RMLVelocitySampleApplication 06_RMLVelocitySampleApplication 
+		07_RMLPositionSampleApplications 08_RMLVelocitySampleApplication
+		DESTINATION bin)
 
